### PR TITLE
Add status quo story about Alan trying to debug an app hang

### PR DESCRIPTION
**Pull request by wesleywiser**
I believe this captures the essence of #76 and may also touch on #75 a bit as some of that info could have been useful to Alan when trying to debug what's going on.

### CODE REVIEW

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (wesleywiser) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:101]: It might be worth expanding this part of the story a bit. I think Alan would probably know that tasks != threads but he also probably don't have a complete understanding of how exactly they work. He may think that there is one thread that loops through all the tasks and tries to wake them up or something.

**Review by nikomatsakis — APPROVED** [review]
This seemed very good to me. I don't have any suggestions for improvement but I'll wait until somebody else gives it a read to merge.

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:13]:
```suggestion
Normally, as a .NET shop, they would have written the entire application in C#, but because of the scalability and latency requirements on their inventory service, they decided to write a microservice in Rust utilizing the `async` features they've heard so much about.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:16]:
```suggestion
After a few cases, a pattern begins to emerge: when a customer tries to buy the last available item, the checkout process hangs forever.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:27]:
```suggestion
He does a quick Google search to see what debugger he should use and decides to go with `gdb` because it is already installed on his system and sounds like it should work.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:28]:
```suggestion
Alan also pulls up a blog post that has a helpful cheatsheet of `gdb` commands since he's not familiar with the debugger.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:31]:
```suggestion
He reproduces the issue a few times in the hope of making it easier to identify the cause of the problem.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:101]:
```suggestion
After looking through the cheatsheet and StackOverflow, he discovers there isn't a way to see which async tasks are waiting to be woken up in the debugger.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:109]:
```suggestion
Disgruntled, Alan begins the arduous, boring task of instrumenting the application in the hope that the logs will be able to help him.
```

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (ibraheemdev) [reply src/vision/status_quo/Alan_tries_to_debug_a_hang.md:101]: Yes, I think this would be helpful. You might also want to clarify whether Alan was using a multi-threaded runtime.

**Review by ibraheemdev — CHANGES_REQUESTED** [review]
Just a couple nits. This looks very good :+1: I do think it would be worth expanding on the section where Alan tries `info threads`.

**Comment by pnkfelix** [comment]
Looks great to me. Great jobs on all those nits you caught, @ibraheemdev !

**Comment by wesleywiser** [comment]
@nikomatsakis Yeah, I think I have a few ideas. Working on it now 🛠️

File: src/vision/status_quo/Alan_tries_to_debug_a_hang.md
Comment (Stupremee) [inline src/vision/status_quo/Alan_tries_to_debug_a_hang.md:117]: The style of the FAQ sections was changed in #118 / #119, using sub headings (###) instead of nested bullet lists.

**Comment by wesleywiser** [comment]
Thanks @Stupremee! I've updated the formatting to match.

**Comment by nikomatsakis** [comment]
This looks great. Thanks @wesleywiser

**Pull request merged**
